test/vonmisesfisher.jl

```diff
@@ -6,8 +6,10 @@ using LinearAlgebra, Test
 using SpecialFunctions
 
 vmfCp(p::Int, κ::Real) = (κ ^ (p/2 - 1)) / ((2π)^(p/2) * besseli(p/2-1, κ))
+logvmfCp(p::Int, κ::Real) = 1/2 * ((p - 2) * log(κ) - 2 * (log(besselix(p/2 - 1, κ)) +  κ) + p * (-log(2π)))
```

Comment:
Personally, I think the amount of brackets makes the code a bit difficult to parse. Maybe use something like
```suggestion
logvmfCp(p::Int, κ::Real) = (p / 2 - 1) * log(κ) - log(besselix(p / 2 - 1, κ) - κ - p / 2 * log(2π)
```
? One could use `log2π` here (from IrrationalConstants/StatsFuns) and avoid some duplicate computations of `p / 2` and `p / 2 - 1` but I don't think it matters since it is only used in the tests.

Reply:
I agree that the whole `VonMisesFisher` implementation could use a bit of a rewrite. So I would not make too many changes right now...

Reply:
This is new and only used in the tests though? So it would be good to ensure that at least the new code is sufficiently clear and simple.